Mastodon/Protocol/Provider/DataSourceFacade+Status.swift

```diff
@@ -144,18 +144,18 @@ extension DataSourceFacade {
 extension DataSourceFacade {
     
     struct MenuContext {
-        let author: ManagedObjectRecord<MastodonUser>? // todo: Remove once IOS-192 is ready
-        let authorEntity: Mastodon.Entity.Account?
+        let author: Mastodon.Entity.Account
         let statusViewModel: StatusView.ViewModel?
         let button: UIButton?
         let barButtonItem: UIBarButtonItem?
     }
     
     @MainActor
-    static func responseToMenuAction(
+    static func responseToMenuAction<T>(
         dependency: UIViewController & NeedsDependency & AuthContextProvider & DataSourceProvider,
         action: MastodonMenu.Action,
-        menuContext: MenuContext
+        menuContext: MenuContext,
+        completion: ((T) -> Void)? = { (param: Void) in }
```

Comment:
Is your question why this method (and the completion) is generic?

Not all menu-actions return a `Mastodon.Entity.Relationship` here. Using the menu you can block a user (returns `Mastodon.Entity.Relationship`), but also a domain returns (`Mastodon.Entity.Empty`) while others like `editStatus` or `translateStatus` do something entirely different.

This was just a first attempt to update a `relationship` IIRC. I consider this WIP and I'm open to other options and ideas!

Reply:
Ah got you, I thought it might be possible to make `responseToMenuAction` async and to just return `T` instead of having to use a closure. But maybe that's not feasible.

Reply:
Let me think about it!

Reply:
I thought about it and although this method is already `async`, it can't return anything, as the important stuff happens in `UIAlertAction`-handlers. To my knowledge these handlers don't support return-values. So, if I want something to happen within those handlers, the only way I can think of is a closure.

I also thought about replacing the completion-closure with `.relationshipChanged`-notification. But as the closure already exists and the return-value is used in `DataSourceProvider+NotificationTableViewCellDelegate` to update the menu in `NotificationView`. Sure, we could replace that with a `UIDeferredMenuElement`, but for now I'd prefer to get this PR merged.